arch: riscv64: smp: get msip base address from dts

yjloong opened the pull request:
In most implements, the msip base address is 0x2000000. But the address is not fixed in all boards.

ycsin commented:
@fkokosinski 
https://github.com/zephyrproject-rtos/zephyr/blob/23fb3b77a2b2cf53eff1697888f443236069b30c/dts/riscv/microchip/microchip-miv.dtsi#L45

cc @manoj153

https://github.com/zephyrproject-rtos/zephyr/blob/23fb3b77a2b2cf53eff1697888f443236069b30c/dts/riscv/efinix/sapphire_soc.dtsi#L62

yjloong commented:
microchip-miv.dtsi and sapphire_soc.dtsi are single hart, which not use SMP.  The problem happens when enabling SMP in multicore environment.

ycsin commented on arch/riscv/core/smp.c:
```diff
@@ -84,7 +84,11 @@ void arch_secondary_cpu_init(int hartid)
-#define MSIP_BASE 0x2000000UL
+#define CLINT_NODE DT_NODELABEL(clint)
+#if !DT_NODE_EXISTS(CLINT_NODE)
+#error "clint label is not defined in the dts."
```
I'd probably just use the term 'devicetree'

```suggestion
#error "clint label is not defined in the devicetree."
```

yjloong replied:
OK.

yjloong replied:
If I can use rebase and 'push -f' to accept the fix?

yjloong replied:
If I can use 'commit --amend' and 'push -f' to accept the fix?

ycsin replied:
What I'd do:
1. `git rebase -i HEAD~2`
2. leave the first commit as `pick`
3. `squash` the second commit
4. keep the commit title/msg/sign-off of the first commit
5. force push

yjloong replied:
AI tells me PR is matching my forked branch. Know more. good.

yjloong replied:
Thanks.

ycsin reviewed (dismissed):
LGTM

yjloong commented:
@ycsin Term "devicetree" is well. 
If I should modify the content in my repo rather than commit changes of yours directly?

yjloong commented:
Last force commit changes:
1. #error "clint label is not defined in the dts."  -->  #error "clint label is not defined in the devicetree."
2. #error "clint label is not defined in the devicetree."  -->  #error "Label 'clint' is not defined in the devicetree."

yjloong commented:
@mgielda @carlocaione @npitre @edersondisouza @kgugala @tgorochowik @katsuster @fkokosinski @con-pax 
Excuse me. If there is someone else has time to review?

con-pax reviewed (dismissed):
LGTM

yjloong commented:
@ycsin Hi, what can I do next?

ycsin commented:
> @ycsin Hi, what can I do next?

Nothing - we are in the freeze period for v4.0.0 release, so it won't be merged just yet.

yjloong commented:
OK. Thanks.

fkokosinski commented on arch/riscv/core/smp.c:
```diff
@@ -84,7 +84,11 @@ void arch_secondary_cpu_init(int hartid)
-#define MSIP_BASE 0x2000000UL
+#define CLINT_NODE DT_NODELABEL(clint)
+#if !DT_NODE_EXISTS(CLINT_NODE)
+#error "Label 'clint' is not defined in the devicetree."
```
I'm not sure about this change. While currently there are no SMP targets in Zephyr that don't have CLINT, this could change in the future. 

I'm aware of at least one out-of-tree (but maintained) port to Linux that doesn't use CLINT - VexRiscv SMP. And there's at least one effort that I'm aware of of porting this target to Zephyr (we already have the non-SMP configuration of it in-tree): https://github.com/zephyrproject-rtos/zephyr/discussions/78053.

fkokosinski replied:
Update: OK, it seems like it is already coupled and https://github.com/zephyrproject-rtos/zephyr/pull/81256 sees to decouple SMP from CLINT

kartben commented:
@yjloong can you please rebase to resolve the merge conflict and then force push to your branch? Thanks!

Commit:
arch: riscv64: smp: get msip base address from dts

In most implements, the msip base address is 0x2000000. But the address
is not fixed in all boards.

Signed-off-by: Yang Jialong <yangjialong@vcore.com>

yjloong commented:
@kartben ok. Review again please.
